Database Schema Types

antiantivirus opened the pull request:
This PR specifies frontend database schemas for all tables we need and also moves to using `&` intersection as this is more common pattern for combining types.

sandreae commented on src/lib/types.ts:
```diff
@@ -554,12 +587,32 @@ type UserRoleAssigned = {
 };
 
+
+type CalendarAccessRequest = {
```
Is this a duplicate with `AccessRequest` below?

sandreae replied:
In which case does `AccessRequest` need `name` and `message` fields?

antiantivirus replied:
I moved this as I thought we don't store this in the db, but (1) I might be wrong there and (2) it could be removed if so?

antiantivirus replied:
And `name` and `message` come from `CalendarAccessRequested` I think. I didn't update this type in this PR.

sandreae reviewed (approved):
All looks good to me except for one comment, thanks!

antiantivirus commented:
@sandreae Should `name` and `message` from `CalendarAccessRequested` also be in our `AccessRequest` db table?

sandreae commented:
> @sandreae Should `name` and `message` from `CalendarAccessRequested` also be in our `AccessRequest` db table?

Yes, I think so :+1:

antiantivirus commented:
Ok great, i updated that and also changed publicKey to from in `AccessRequest` so its clearer and follow same structure as `AccessResponse` + fixed up the `access.ts` so its all working now. 

This should be good to merge now.

antiantivirus commented:
> Yes, I think so 👍

Ok great, i updated that and also changed publicKey to from in AccessRequest so its clearer and follow same structure as AccessResponse + fixed up the access.ts so its all working now.

This should be good to merge now.
